Sources/Datadog/Core/System/MobileDevice.swift

```diff
@@ -107,22 +133,48 @@ internal class MobileDevice {
         }
     }
 
-    #else
+    #elseif os(tvOS)
+
     convenience init(
-        uiDevice: UIDevice = .current,
-        processInfo: ProcessInfo = .processInfo,
-        notificationCenter: NotificationCenter = .default
+        model: String,
+        uiDevice: UIDevice,
+        processInfo: ProcessInfo,
+        notificationCenter: NotificationCenter
     ) {
-        // iOS Simulator - battery monitoring doesn't work on tvOS nor Simulator, so return "always OK" value
         self.init(
-            model: uiDevice.model,
+            name: uiDevice.model,
+            model: model,
             osName: uiDevice.systemName,
             osVersion: uiDevice.systemVersion,
+            // Battery monitoring doesn't work on tvOS, so return "always OK" value:
             enableBatteryStatusMonitoring: {},
             resetBatteryStatusMonitoring: {},
             currentBatteryStatus: { BatteryStatus(state: .full, level: 1, isLowPowerModeEnabled: false) }
         )
     }
+
+    convenience init() {
+        let processInfo = ProcessInfo.processInfo
+        let device = UIDevice.current
+        let model: String
+
+        #if !targetEnvironment(simulator)
+        // Real tvOS device
+        model = (try? Sysctl.getModel()) ?? device.model
+        #else
+        // tvOS Simulator
+        let simulatorModel = processInfo.environment["SIMULATOR_MODEL_IDENTIFIER"] ?? device.model
+        model = "\(simulatorModel) Simulator"
+        #endif
+
+        self.init(
+            model: model,
+            uiDevice: device,
+            processInfo: processInfo,
+            notificationCenter: .default
+        )
+    }
```

Comment:
These new initialisers in `MobileDevice` are added for init consistency on both `iOS` and `tvOS` (both define matching `init()`). This and the entire `MobileDevice` class should be however a subject to change in V2. Since we added `tvOS` support, it should no longer be "mobile device" and we should really split its responsibilities. For example, the cohesion of this class is very low - battery reasoning and management is only implemented for `iOS` and we use mock values for `tvOS` just to "make it work".

Reply:
Sounds good!